main.go

```diff
@@ -1,5 +1,121 @@
 package main
 
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/db"
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/forman"
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/reconciler"
+	"go.uber.org/zap"
+	"gopkg.in/yaml.v2"
+)
+
+const configFileEnvVarName = "FLEET_RECONCILER_CONFIG"
 
 func main() {
+	//logger := logr.Logger{}
+	ctx := context.Background()
+	configFilePath := os.Getenv(configFileEnvVarName)
+	if configFilePath == "" {
+		log.Fatalf("%q environment variable not set", configFileEnvVarName)
+	}
+	// parse configuration from yaml file
+	conf, err := loadAndValidateConfig(configFilePath)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	zapConfig := zap.Config{
+		Encoding:         "json",
+		EncoderConfig:    zap.NewProductionEncoderConfig(),
+		OutputPaths:      []string{"stdout"},
+		ErrorOutputPaths: []string{"stderr"},
+	}
+	logger, err := zapConfig.Build()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	connString := getConnectionString(conf)
```

Comment:
For the sake of separation of concerns, and as requested in previous commits. please send the separate bits to this function:  
```suggestion
	connString := getConnectionString(conf.DB.Host, conf.DB.Port, conf.DB.User, conf.DB.Password, conf.DB.Name)
```